pkg/controller/azure/compute/claim.go

```diff
@@ -67,21 +67,31 @@ func ConfigureAKSCluster(_ context.Context, cm resource.Claim, cs resource.Class
 		return errors.Errorf("expected resource claim %s to be %s", cm.GetName(), computev1alpha1.KubernetesClusterGroupVersionKind)
 	}
 
-	rs, csok := cs.(*corev1alpha1.ResourceClass)
+	rs, csok := cs.(*v1alpha1.AKSClusterClass)
 	if !csok {
-		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), corev1alpha1.ResourceClassGroupVersionKind)
+		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), v1alpha1.AKSClusterClassGroupVersionKind)
 	}
 
 	i, mgok := mg.(*v1alpha1.AKSCluster)
 	if !mgok {
 		return errors.Errorf("expected managed resource %s to be %s", mg.GetName(), v1alpha1.AKSClusterGroupVersionKind)
 	}
 
-	spec := v1alpha1.NewAKSClusterSpec(rs.Parameters)
+	spec := &v1alpha1.AKSClusterSpec{
+		ResourceSpec: corev1alpha1.ResourceSpec{
+			ReclaimPolicy: corev1alpha1.ReclaimRetain,
+		},
+		AKSClusterParameters: rs.SpecTemplate.AKSClusterParameters,
+	}
+
+	// NOTE(hasheddan): consider moving defaulting to either CRD or managed reconciler level
```

Comment:
There are a few of these notes about defaulting properties.  It would be good to get that captured in an issue.

Reply:
Agreed. I think it may be good to call it out as part of the "high fidelity" issues for the infra stacks epics. Here is an example with Azure issue #648.

Reply:
I expect that there won't e time in v0.3 to address defaulting across all the resource types as well, can we open a general issue to track investigating and implementing a good solution for defaulting post v0.3?

Reply:
Absolutely!